src/components/PieChart/PieChart.jsx

```diff
@@ -0,0 +1,344 @@
+			onMouseOut: _.noop,
+
+			xAxisField: 'x',
+			xAxisFormatter: _.identity,
+
+			yAxisField: 'y',
+			yAxisFormatter: _.identity,
+		};
+	},
+
+	handleMouseOver(index, event) {
+		this.props.onMouseOver(index, {
+			props: this.props,
+			event,
+		});
+	},
+
+	handleMouseOut(event) {
+		this.props.onMouseOut({
+			props: this.props,
+			event,
+		});
+	},
+
+	render() {
+		const {
+			className,
+			height,
+			width,
+			margin: marginOriginal,
+			data,
+			hasToolTips,
+			palette,
+			colorMap,
+			isDonut,
+			ToolTip: toolTipProps,
+
+			isHovering,
+			hoveringIndex,
+
+			xAxisField,
+			xAxisFormatter,
+
+			yAxisField,
+			yAxisFormatter,
+
+			...passThroughs,
+		} = this.props;
+
+		const margin = {
+			...PieChart.MARGIN,
+			...marginOriginal,
+		};
+
+		// TODO: Consider displaying something specific when there is no data,
+		// perhaps a loading indicator.
+		if (_.isEmpty(data)) {
+			return (
+				<svg
+					{..._.omit(passThroughs, ['onMouseOver', 'onMouseOut'])}
+					className={cx(className, '&')}
+					width={width}
+					height={height}
+				/>
+			);
+		}
+
+		const innerWidth = width - margin.left - margin.right;
+		const innerHeight = height - margin.top - margin.bottom;
+
+		const outerRadius = Math.min(innerWidth, innerHeight) / 2;
+
+		const arc = d3Shape.arc()
+			.innerRadius(isDonut ? outerRadius - PieChart.DONUT_WIDTH : INNER_RADIUS)
+			.outerRadius(outerRadius);
+
+		// Useful for capturing hovers when we're in donut mode
+		const arcFull = d3Shape.arc()
+			.innerRadius(0)
+			.outerRadius(outerRadius);
+
+		const pie = d3Shape
+			.pie()
+			.sort(null); // needed to put the slices in proper order
+		const pieData = pie(_.map(data, yAxisField));
+
+		const renderPie = (
+			<g transform={`translate(${margin.left}, ${margin.top})`}>
+				<g transform={`translate(${innerWidth / 2}, ${innerHeight / 2})`}>
+					{_.map(pieData, (pieDatum, index) => (
+						<g
+							key={index}
+							className={cx('&-slice-group', {
+								'&-slice-group-is-hovering': isHovering && hoveringIndex === index,
+							})}
+						>
+							<Line
+								key={index}
+								className={cx('&-slice')}
+								d={arc(pieDatum)}
+								color={_.get(colorMap, data[index][xAxisField], palette[index % palette.length])}
+								hasStroke={false}
+								transform={`scale(${isHovering && hoveringIndex === index ? HOVER_SCALE : 1})`}
+							/>
+
+							{/* This hidden path is useful for capturing hovers when we're in donut mode */}
+							<path
+								className={cx('&-slice-hover')}
+								d={arcFull(pieDatum)}
+								transform={`scale(${HOVER_SCALE})`}
+								onMouseOver={_.partial(this.handleMouseOver, index)}
+								onMouseOut={hasToolTips ? _.noop : this.handleMouseOut}
+							/>
+						</g>
+					))}
+				</g>
+			</g>
+		);
+
+		return (
+			<svg
+				{..._.omit(passThroughs, ['onMouseOver', 'onMouseOut'])}
+				className={cx(className, '&')}
+				width={width}
+				height={height}
+			>
+				{hasToolTips ?
```

Comment:
Though you might still be right, I might be able to move `renderPie` into the target and do what you said.